### PR TITLE
irmin-tezos (test_gc): Add `check_async_unlinked` that checks if a file has been unlinked asynchronously.

### PR DESCRIPTION
**Pull request by adatario**
This fixes flakey test failures introduced by #2221 caused by files still existing that are being unlinked asynchronously (see https://github.com/mirage/irmin/pull/2227#issuecomment-1486707570).

The introduced `check_asyn_unlinked` function takes a timeout argument and checks if the file has been unlinked within the specified timeout period.

Tested locally:

- [x] Use negative timeout to check failure
- [ ] Use small timeout to reproduce condition that should be catched.

I've run the tests multiple times and can not reproduce the conditions observed in the CI. Still, I believe this should fix the issue.

### CODE REVIEW

**Review by art-w — APPROVED** [review]
Thanks! I agree, it's nice that your solution still tests the real async code... Let's see if it works, we can do something more fancy if we discover that the test is still flaky :)

File: test/irmin-pack/test_gc.ml [hunk 1]
```diff
@@ -265,6 +265,13 @@ module type Gc_backend = sig
   val check_removed : t -> S.commit -> string -> unit Lwt.t
 end
 
+let rec check_async_unlinked ?(timeout = 3.141) file =
```
Comment (metanivek) [inline test/irmin-pack/test_gc.ml:268]: :smirk_cat: 
```suggestion
let rec check_async_unlinked ?(timeout = Float.pi) file =
```

**Review by metanivek — APPROVED** [review]
Reasonable solution. We can always re-visit if it turns out to be flaky (my eye is on any upcoming opam release :sweat_smile:).

Left a joke suggestion (but feel free to implement!)

File: test/irmin-pack/test_gc.ml (hunk 1 above)
Comment (adatario) [reply test/irmin-pack/test_gc.ml:268]: :) I'll keep that for a future update.

**Pull request merged**